bridgetown-core/lib/bridgetown-core/tags/webpack_path.rb

```diff
@@ -2,37 +2,77 @@
 
 module Bridgetown
   module Tags
+    # A helper class to help find the path to webpack asset inside of a webpack
+    # manifest file.
     class WebpackPath < Liquid::Tag
       include Bridgetown::Filters::URLFilters
 
-      def initialize(tag_name, asset_type, tokens)
+      # @param tag_name [String] Name of the tag
+      # @param asset_type [String] The type of asset to parse (js, css)
+      # @param options [Hash] An options hash
+      # @return WebpackPath
+      # @see {https://www.rdoc.info/github/Shopify/liquid/Liquid/Tag#initialize-instance_method}
+      def initialize(tag_name, asset_type, options)
         super
 
         # js or css
         @asset_type = asset_type.strip
       end
 
+      # Render the contents of a webpack manifest file
+      # @param context [String] Root directory that contains the manifest file
+      #
+      # @return [String] Returns "MISSING_WEBPACK_MANIFEST" if the manifest
+      # file isnt found
+      # @return [nil] Returns nil if the asset isnt found
+      # @return [String] Returns the path to the asset if no issues parsing
+      #
+      # @raise [WebpackAssetError] if unable to find css or js in the manifest
+      # file
       def render(context)
         @context = context
         site = context.registers[:site]
 
+        manifest_file = site.in_root_dir(".bridgetown-webpack", "manifest.json")
+
+        parse_manifest_file(manifest_file)
+      end
+
+      private
+
+      def parse_manifest_file(manifest_file)
+        return "MISSING_WEBPACK_MANIFEST" unless File.exist?(manifest_file)
+
+        manifest = JSON.parse(File.read(manifest_file))
         frontend_path = relative_url("_bridgetown/static")
 
-        manifest_file = site.in_root_dir(".bridgetown-webpack", "manifest.json")
-        if File.exist?(manifest_file)
-          manifest = JSON.parse(File.read(manifest_file))
-          if @asset_type == "js"
-            js_path = manifest["main.js"].split("/").last
-            [frontend_path, "js", js_path].join("/")
-          elsif @asset_type == "css"
-            css_path = manifest["main.css"].split("/").last
-            [frontend_path, "css", css_path].join("/")
-          else
-            Bridgetown.logger.error("Unknown Webpack asset type", @asset_type)
-            nil
-          end
-        else
-          "MISSING_WEBPACK_MANIFEST"
+        known_assets = %w(js css)
+
+        if known_assets.include?(@asset_type)
+          asset_path = manifest["main.#{@asset_type}"]
+
+          raise_webpack_asset_error(@asset_type) if asset_path.nil?
+
+          asset_path = asset_path.split("/").last
+          return [frontend_path, @asset_type, asset_path].join("/")
+        end
+
+        Bridgetown.logger.error("Unknown Webpack asset type", @asset_type)
+        nil
+      end
```

Comment:
Your implementation with `know_assets` is much better!

- I'd personally get rid of the `parse_manifest_file`  private method. It seems like it's moving the main _body_ of the work from the `render` method (where it should be) to a secondary place (`parce_manifest_file`) needlessly. So when reading the code, I'd actually focus on this private method rather than the main one `render`. 

There are details of the method that could potentially extracted to private methods such as the json.parse(file.read) .. but that's your preference. I don't think there's scary need to do this.

- The explicit return of nil is a bit off, maybe you'd rather shorten the path like you did with the other one with:

```ruby
unless known_assets.include?(@asset_type)
  Bridgetown.logger.error("Unknown Webpack asset type", @asset_type)
  return nil
end
```

- I'd put `known_assets` in a constant at the top. _handled_asset_types_  or _handled_assets_ might be a bit cleaner, not sure, your call.

Reply:
@mtomov Given the work we'll need to do in #100 to extract this parsing logic out to a neutral utility method, I think this is fine for now. I like the suggestion about moving the assets list to a constant.